<a href="https://colab.research.google.com/github/Aldistwn29/Matakuliah_Data_Science/blob/main/statistika_inferensial.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Uji Hipotesis**

In [ ]:
# import library
import numpy as np
from scipy import stats

# asumsi data penjualan dari kedua cabang
data_cab1 =  [150, 160, 155, 170, 145, 152, 158, 167, 155, 160, 165, 150, 170,
              162, 159, 155, 161, 152, 158, 165, 171, 156, 157, 162, 159, 155, 164, 166, 158, 163]
data_cab2 =  [148, 155, 152, 160, 149, 151, 158, 157, 154, 153, 155, 149, 157,
              161, 150, 152, 155, 148, 151, 157, 160, 154, 153, 150, 156, 158, 159, 154, 150, 151]

# melakukan uji t, 2 sampel
t_stat, p_value = stats.ttest_ind(data_cab1, data_cab2)

# menampilkan hasil
print(f'Nilai t-stats:{t_stat:.2f}')
print(f'Nilai p-value:{p_value:.2f}')

# nilai keputusan
alpha = 0.05

if p_value < alpha:
    print('terdapat perbedaan siginifikan dalam rata-rata penjualan cabang 1 dan cabang 2')
else:
    print('tidak ada perbedaan signifikan dalam rata-rata penjualan cabang 1 dan cabang 2')

Nilai t-stats:4.02
Nilai p-value:0.00
terdapat perbedaan siginifikan dalam rata-rata penjualan cabang 1 dan cabang 2


**Interpertasi:**\
Maka H0 di tolak, kita tidak punya banayak terdapat perbedaan signifikan dalam rata-rata penjualan cabang 1 dan cabang 2

In [ ]:
import math


def rata_rata(data):
  n = len(data)

  mean = sum(data)/n
  return mean

def variansi(data):
  n = len(data)
  mean = rata_rata(data)
  variansi = sum((x - mean) ** 2 for x in data)/n
  return variansi

def uji_t(data1, data2):

  mean1 = rata_rata(data1)
  mean2 = rata_rata(data2)

  variansi1 = variansi(data1)
  variansi2 = variansi(data2)

  n1 = len(data1)
  n2 = len(data2)

  stat_t = (mean1 - mean2) / math.sqrt((variansi1/n1 + variansi2/n2))
  return stat_t

In [ ]:
uji_t(data_cab1, data_cab2)

4.091183661346218

# **Interval Kepercayaan**

In [12]:
import numpy as np
from scipy import stats

# Data set
data_survei_1 = [500, 520, 510, 505, 498, 502, 515, 490, 485, 500]
data_survei_2 = [480, 495, 470, 460, 485, 475, 465, 485, 475, 480]

# Menghitung rata-rata
mean1 = np.mean(data_survei_1)
mean2 = np.mean(data_survei_2)

# Menghitung standar deviasi
std1 = np.std(data_survei_1, ddof=1)
std2 = np.std(data_survei_2, ddof=1)

# Ukuran sample
n1 = len(data_survei_1)
n2 = len(data_survei_2)

# Menghitung standar error
se = np.sqrt((std1 ** 2 / n1) + (std2 ** 2 / n2))


# menentukan tingkat kepercayaan
confidence = 0.95
alpha = 1 - confidence

# nilai t kritis untuk tingkat kepercayaan 95%
t_critical = stats.t.ppf(1 - alpha/2, df= n1 + n2 - 2)

# margin of error
moe = t_critical * se

# Perbedaan rata-rata
mean_did = mean1 - mean2

# Interval kepercayaan
lower_bound = mean_did - moe
upper_bound = mean_did + moe


print("rata-rata data1: ", mean1)
print("rata-rata data2: ", mean2)
print("standar deviansi data1: ", std1)
print("standar deviansi data1: ", std2)
print("Ukuran Sempel n1: ", n1)
print("Ukuran Sempel n2: ", n2)
print("Standar error: ", se)
print("Nilai t-kritis: ", t_critical)
print("Margin of error: ", moe)
print("Perbedaan rata-rata: ",mean_did)
print(f"Interval kepercayaan: {lower_bound}, {upper_bound }")

rata-rata data1:  502.5
rata-rata data2:  477.0
standar deviansi data1:  10.64842189664219
standar deviansi data1:  10.327955589886445
Ukuran Sempel n1:  10
Ukuran Sempel n2:  10
Standar error:  4.6910079466523555
Nilai t-kritis:  2.10092204024096
Margin of error:  9.855441986067424
Perbedaan rata-rata:  25.5
Interval kepercayaan: 15.644558013932576, 35.355441986067426


**Interpertasi**: \
- Dari hasil yang di dapatkan di atas dengan tingkat kepercayaan 95% dan interval kepercayaanya [15.644558013932576,35.355441986067426], dengan selisih rata-ratanya :  25.5.
- Mecakup nol: ini berarti terdapat perbedaan yang signifikan antar rata-rata kepuasan penguna sebelum dan sebelum pengenalan fitur baru. Artinya perusahaan percaya bahawa pengenalan fitur baru berpengaruh terhadap kepuasan pada pelanggan.
